src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudServiceTest.kt

```diff
@@ -1099,6 +1108,49 @@ class CosmasGoogleCloudServiceTest {
             addPatchToService(patch)
             commit()
         }
+        return fakeStorage
+    }
+
+    @Test
+    fun versionsNotExistFromMemory() {
+        val month = 24 * 60 * 60 * 1000
+        val clock = mock(Clock::class.java)
+        Mockito.`when`(clock.millis())
+                .thenReturn(0L)
+                .thenReturn(1L)
+                .thenReturn(month + 1L)
+                .thenReturn(month + 2L)
+                .thenReturn(0L)
```

Comment:
Break these thenReturns into two blocks? If I get it right, first few values are needed for addManyVersions, while last few are used in getVersionsList ? In this case your can move the tail after addManyVersions call

Reply:
first 4 values are used in `addManyVersions`, next 4 in commit inside `addManyVersions`, and only the last one is used in `getVersionsList`, so I splitted it like this